### PR TITLE
style: migrate to eslint

### LINKED ISSUE #4483: Migrate from tslint to eslint
`tslint` is deprecated in favor of `eslint`. We should switch over.

See [this blog post](https://medium.com/palantir/tslint-in-2019-1a144c2317a9) and [this issue](https://github.com/palantir/tslint/issues/4534).
Edit: also [this](https://code.visualstudio.com/api/advanced-topics/tslint-eslint-migration)

### PR DESCRIPTION
**Pull request by Foo-x**
**What this PR does / why we need it**:

- install eslint
- create eslint config with `npx tslint-to-eslint-config`
    - ignored some rules to conform to current code
- update existing config
    - npm scripts
    - vscode extensions
- fix code to satisfy the rules
- uninstall tslint
- remove tslint.json

**Which issue(s) this PR fixes**

Closes #4483 

**Special notes for your reviewer**:

I have ignored some lines difficult to immediately satisfy the eslint rules by `eslint-disable-next-line`.

### CODE REVIEW

File: src/actions/plugins/surround.ts [hunk 1]
```diff
@@ -120,7 +111,16 @@ class YankSurroundOperator extends SurroundOperator {
...
+    };
+    // then collect ranges for all cursors
+    const multicursorIndex = this.multicursorIndex ?? 0;
+    vimState.surround!.edges.push(getYankRanges());
+    vimState.cursorStartPosition = start;
+    // when called from visual operator, use end for stop to keep visual selection
+    vimState.cursorStopPosition = vimState.currentMode === Mode.Visual ? end : start;
+    await vimState.setCurrentMode(Mode.SurroundInputMode);
+
+    return;
```
Comment (J-Fields) [inline src/actions/plugins/surround.ts:123]: This return is now (more obviously) redundant

Comment (Foo-x) [reply src/actions/plugins/surround.ts]: fixed.

**Review by J-Fields — APPROVED** [review]
Thank you!

**Review by laligarcia — COMMENTED** [review]
I want to be receiving a verify Whatsapp codes from my Whatsapp number

**Pull request merged**